### PR TITLE
WEBAPP-174: Adjust linting

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

**Comment by klinzo** [comment]
Relates to issue 274 right? https://integreat.atlassian.net/browse/WEBAPP-274

**Comment by maxammann** [comment]
@klinzo oops yep, will rewrite the branch history

File: src/flowTypes.js [hunk 1]
```diff
@@ -9,34 +9,34 @@ import EventModel from './modules/endpoint/models/EventModel'
...
-  +uiDirection: UiDirection
+  +uiDirection: UiDirectionType
 }
 
-export type EndpointParams = {city?: string, language?: string, url?: string}
-export type MapParamsToUrl = (params: EndpointParams) => string
-export type MapResponse = (json: any, params: EndpointParams) => PayloadData
+export type EndpointParamsType = {city?: string, language?: string, url?: string}
+export type MapParamsToUrlType = (params: EndpointParamsType) => string
+export type MapResponseType = (json: any, params: EndpointParamsType) => PayloadDataType
```
Comment (Schedulaar) [inline src/flowTypes.js]: When should these have suffix `Type`? Only these explicitly defined as `type`? Normal classes (like all our react components) are also valid flow `type`s afaik, and we obviously don't append `Type` to those...
At least it's a bit odd to me, that we need to append this suffix...

File: src/modules/app/containers/App.js [hunk 2]
```diff
@@ -10,24 +12,30 @@ import PlatformProvider from '../../platform/containers/PlatformProvider'
 import routesMap from '../routesMap'
 import Switcher from './Switcher'
 import theme from '../constants/theme'
+import { Store } from 'redux'
+
+export interface PropsType {
```
Comment (Schedulaar) [inline src/modules/app/containers/App.js]: Do we need empty PropTypes now?
Should we always export them?

File: src/modules/i18n/containers/I18nProvider.js [hunk 3]
```diff
@@ -86,22 +113,38 @@ export class I18nProvider extends React.Component {
...
+          style={{
+            direction: RTL_LANGUAGES.includes(this.state.language)
+              ? 'rtl'
+              : 'ltr'
+          }}
+        >
+          <ReactHelmet>
+            {lateef && (
+              <link href='/fonts/lateef/lateef.css' rel='stylesheet' />
+            )}
```
Comment (Schedulaar) [inline src/modules/i18n/containers/I18nProvider.js]: Do we force the parantheses `(` and `)` around jsx components? I'd suggest not to do that.

File: src/routes/events/containers/EventsPage.js [hunk 4]
```diff
@@ -51,15 +51,15 @@ export class EventsPage extends React.Component<Props> {
...
-  city: state.location.payload.city,
-  eventId: state.location.payload.eventId,
-  events: state.events.data,
-  cities: state.cities.data
+const mapStateTypeToProps = (stateType: StateType) => ({
+  language: stateType.location.payload.language,
+  city: stateType.location.payload.city,
+  eventId: stateType.location.payload.eventId,
+  events: stateType.events.data,
+  cities: stateType.cities.data
```
Comment (Schedulaar) [inline src/routes/events/containers/EventsPage.js]: `stateType` looks odd, as its not a type. Should be `state`.

File: src/routes/landing/components/FilterableCitySelector.js [hunk 5]
```diff
@@ -8,34 +9,46 @@ import CityModel from 'modules/endpoint/models/CityModel'
...
+
+type StateType = {
+  filterText: string
+}
 
-  constructor (props) {
+export class FilterableCitySelector extends React.Component<
+  PropsType,
+  StateType
+> {
```
Comment (Schedulaar) [inline src/routes/landing/components/FilterableCitySelector.js]: maybe we need to adjust our maximum line width, so we don't need too much of these line breaks.

File: src/routes/search/containers/SearchPage.js [hunk 6]
```diff
@@ -10,22 +9,22 @@ import CategoriesMapModel from 'modules/endpoint/models/CategoriesMapModel'
...
+type PropsType = {
   categories: CategoriesMapModel,
   cities: Array<CityModel>,
   city: string,
-  t: I18nTranslate
+  t: I18nTranslateType
 }
 
-type LocalState = {
+type LocalStateTypeType = {
```
Comment (Schedulaar) [inline src/routes/search/containers/SearchPage.js]: LocalStateTypeType? :)

Comment (maxammann) [reply src/routes/search/containers/SearchPage.js]: Oops, will fix, thanks :)

File: src/routes/landing/components/FilterableCitySelector.js (hunk 5 above)
Comment (maxammann) [reply src/routes/landing/components/FilterableCitySelector.js]: Yes, that looks odd. If we change a eslint rule I won't do it in this task.

But I'll open an issue for this.

File: src/routes/events/containers/EventsPage.js (hunk 4 above)
Comment (maxammann) [reply src/routes/events/containers/EventsPage.js]: Yep, thank :+1:

File: src/modules/i18n/containers/I18nProvider.js (hunk 3 above)
Comment (maxammann) [reply src/modules/i18n/containers/I18nProvider.js]: I think we do not force them.

File: src/modules/app/containers/App.js (hunk 2 above)
Comment (maxammann) [reply src/modules/app/containers/App.js]: I think we should not export them. Not sure about the empty ones.

File: src/flowTypes.js (hunk 1 above)
Comment (maxammann) [reply src/flowTypes.js]: That is a linting rule in neuburg-backend. Open for discussion I suppose.

File: src/modules/i18n/containers/I18nProvider.js (hunk 3 above)
Comment (klinzo) [reply src/modules/i18n/containers/I18nProvider.js]: There are quite some in the changes above though.

File: src/modules/layout/components/HeaderDropDown.js [hunk 7]
```diff
@@ -1,23 +1,32 @@
...
  * Header. Once the user clicks outside, the node is hidden again. Additionally, the inner node gets a
  * closeDropDownCallback through its props to close the dropDown and hide itself.
  */
-class HeaderDropDown extends React.Component {
-  static propTypes = {
-    iconSrc: PropTypes.string.isRequired,
-    children: PropTypes.element.isRequired
-  }
+class HeaderDropDown extends React.Component<PropsType, StateType> {
+  toggleDropDown: Function
```
Comment (klinzo) [inline src/modules/layout/components/HeaderDropDown.js]: What type of function?

Comment (maxammann) [reply src/modules/layout/components/HeaderDropDown.js]: All of these are marked as warnings in flow. Can be fixed later. There are a lot of similar problems

**Pull request merged**